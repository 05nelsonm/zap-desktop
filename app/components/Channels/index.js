export ChannelCapacity from './ChannelCapacity'
export ChannelMoreButton from './ChannelMoreButton'
export ChannelStatus from './ChannelStatus'
export ChannelSummaryList from './ChannelSummaryList'
export ChannelSummaryListItem from './ChannelSummaryListItem'
