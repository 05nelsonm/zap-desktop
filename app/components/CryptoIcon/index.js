import CryptoIcon from './CryptoIcon'

export default CryptoIcon
