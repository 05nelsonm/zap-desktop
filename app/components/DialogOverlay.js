import styled from 'styled-components'
import { Flex } from 'rebass'

const Overlay = styled(Flex)`
  position: absolute;
  z-index: 99999;
  background-color: rgba(255, 255, 255, 0.25);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

export default Overlay
