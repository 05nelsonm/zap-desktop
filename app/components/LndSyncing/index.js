import LndSyncing from './LndSyncing'

export default LndSyncing
