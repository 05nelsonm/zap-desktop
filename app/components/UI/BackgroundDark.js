import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundDark
 * @example
 * <BackgroundDark />
 */
class BackgroundDark extends React.Component {
  render() {
    return <Box bg="darkestBackground" color="primaryText" {...this.props} />
  }
}

export default BackgroundDark
