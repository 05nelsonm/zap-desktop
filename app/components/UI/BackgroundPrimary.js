import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundPrimary
 * @example
 * <BackgroundPrimary />
 */
class BackgroundPrimary extends React.Component {
  render() {
    return <Box bg="primaryColor" color="primaryText" {...this.props} />
  }
}

export default BackgroundPrimary
