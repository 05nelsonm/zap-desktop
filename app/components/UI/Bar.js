import styled from 'styled-components'
import { variant, space, height } from 'styled-system'

const bars = variant({
  key: 'bars',
})

const Bar = styled.hr`
  height: 1px;
  margin: 0;
  padding: 0;
  border: 0;
  background-color: ${props => props.theme.colors.primaryText};
  ${space}
  ${height}
  ${bars}
`

Bar.defaultProps = {
  variant: 'normal',
}

Bar.displayName = 'Bar'

export default Bar
