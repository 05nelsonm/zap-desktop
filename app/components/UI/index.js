export ActionBar from './ActionBar'
export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export Button from './Button'
export Checkbox from './Checkbox'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export Notification from './Notification'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Value from './Value'
export Wizard from './Wizard'
