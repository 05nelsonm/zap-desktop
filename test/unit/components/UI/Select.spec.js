import React from 'react'
import { Form } from 'informed'
import renderer from 'react-test-renderer'
import { dark } from 'themes'
import { ThemeProvider } from 'styled-components'
import { Select } from 'components/UI'

describe('component.UI.Toggle', () => {
  it('should render correctly', () => {
    const selectItems = [
      { value: '- Please select -', key: '' },
      { value: 'Apple', key: 'apple' },
      { value: 'Pear', key: 'pear' },
      { value: 'Orange', key: 'orange' },
      { value: 'Grape', key: 'grape' },
      { value: 'Banana', key: 'banana' },
    ]
    const tree = renderer
      .create(
        <ThemeProvider theme={dark}>
          <Form>
            <Select field="name" items={selectItems} />
          </Form>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
