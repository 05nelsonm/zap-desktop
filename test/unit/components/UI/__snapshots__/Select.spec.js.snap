// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Toggle should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  padding: 16px;
  color: #ffffff;
  background-color: transparent;
  border: 1px solid;
  border-color: #959595;
  border-radius: 5px;
  font-family: Roboto,system-ui,sans-serif;
  font-size: 13px;
  font-weight: 300;
  width: 100%;
  opacity: inherit;
  outline: none;
  border-color: #959595;
}

.c3:not([readOnly]):not([disabled]):focus {
  border-color: #fd9800;
}

.c3::-webkit-search-decoration:hover,
.c3::-webkit-search-cancel-button:hover {
  cursor: pointer;
}

.c4 {
  margin-left: -26px;
  width: 10px;
  pointer-events: none;
  color: #959595;
}

.c1 input {
  cursor: pointer;
  color: transparent;
  text-shadow: 0 0 0 #ffffff;
}

<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-labelledby="downshift-0-label"
    aria-owns={null}
    role="combobox"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="c0"
    >
      <div
        className="c1 c2"
      >
        <div
          className="c0"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            autoComplete="off"
            autoFocus={false}
            className="c3"
            color="primaryText"
            disabled={false}
            fontFamily="sans"
            fontSize="m"
            fontWeight="light"
            id="downshift-0-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            placeholder="Please select"
            required={false}
            value=""
            width={1}
          />
        </div>
      </div>
      <div
        className=""
      >
        <svg
          className="c4"
          height="1em"
          viewBox="0 0 8 4"
          width={10}
        >
          <path
            d="M1 0l3 4 3-4"
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>
</form>
`;
